tck/cmd/server.go

```diff
@@ -29,8 +43,16 @@ func main() {
 	if port == "" {
 		port = "80"
 	}
-	err := http.ListenAndServe(":"+port, nil)
-	if err != nil {
-		panic(err)
-	}
+	fmt.Println("Server is listening on port: " + port)
+	go func() {
+		http.ListenAndServe(":"+port, nil)
+	}()
+
+	signalCh := make(chan os.Signal, 1)
+	signal.Notify(signalCh, os.Interrupt, syscall.SIGTERM)
+	sig := <-signalCh
```

Comment:
Could add timeout context of 5 sec to the server so that any ongoing requests are finished before shutting down the server.

```go
    ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
	defer cancel()

	if err := server.Shutdown(ctx); err != nil {
		log.Printf("Graceful shutdown failed: %v\n", err)
		os.Exit(1)
	}
```